private_transactions/src/lib.rs

```diff
@@ -0,0 +1,782 @@
+			}).collect()),
+			ethabi::Token::Array(signatures.iter().map(|s| ethabi::Token::FixedBytes(s.r().to_vec())).collect()),
+			ethabi::Token::Array(signatures.iter().map(|s| ethabi::Token::FixedBytes(s.s().to_vec())).collect()),
+		];
+
+		function.encode_input(&tokens).expect("Input is always valid")
+	}
+
+	/// Returns the key from the key server associated with the contract
+	pub fn contract_key_id(&self, contract_address: &Address) -> Result<H256, PrivateTransactionError> {
+		//current solution uses contract address extended with 0 as id
+		let contract_address_extended: H256 = contract_address.into();
+
+		Ok(H256::from_slice(&contract_address_extended))
+	}
+
+	/// Create encrypted public contract deployment transaction.
+	pub fn public_creation_transaction(&self, block: BlockId, source: &SignedTransaction, validators: &[Address], gas_price: U256) -> Result<Transaction, PrivateTransactionError> {
+		if let &Action::Call(_) = &source.action {
+			return Err(PrivateTransactionError::BadTransactonType.into());
+		}
+		let sender = source.sender();
+		let state = self.client.state_at(block).ok_or(PrivateTransactionError::StatePruned)?;
+		let nonce = state.nonce(&sender)?;
+		let executed = self.execute_private(source, TransactOptions::with_no_tracing(), block)?;
+		let gas: u64 = 650000 +
+			validators.len() as u64 * 30000 +
+			executed.code.as_ref().map_or(0, |c| c.len() as u64) * 8000 +
+			executed.state.len() as u64 * 8000;
+		Ok(Transaction {
+			nonce: nonce,
+			action: Action::Create,
+			gas: gas.into(),
+			gas_price: gas_price,
+			value: source.value,
+			data: Self::generate_constructor(validators, executed.code.unwrap_or_default(), executed.state)
+		})
+	}
+
+	/// Create encrypted public contract deployment transaction. Returns updated encrypted state.
+	fn execute_private_transaction(&self, block: BlockId, source: &SignedTransaction) -> Result<Bytes, PrivateTransactionError> {
+		if let &Action::Create = &source.action {
+			return Err(PrivateTransactionError::BadTransactonType.into());
+		}
+		let result = self.execute_private(source, TransactOptions::with_no_tracing(), block)?;
+		Ok(result.state)
+	}
+
+	/// Create encrypted public transaction from private transaction.
+	pub fn public_transaction(&self, state: Bytes, source: &SignedTransaction, signatures: &[Signature], nonce: U256, gas_price: U256) -> Result<Transaction, PrivateTransactionError> {
+		let gas: u64 = 650000 + state.len() as u64 * 8000 + signatures.len() as u64 * 50000;
+		Ok(Transaction {
+			nonce: nonce,
+			action: source.action.clone(),
+			gas: gas.into(),
+			gas_price: gas_price,
+			value: 0.into(),
+			data: Self::generate_set_state_call(signatures, state)
+		})
+	}
+
+	/// Call into private contract.
+	pub fn private_call(&self, block: BlockId, transaction: &SignedTransaction) -> Result<Executed, PrivateTransactionError> {
+		let result = self.execute_private(transaction, TransactOptions::with_no_tracing(), block)?;
+		Ok(result.result)
```

Comment:
no need for `?`. one line is enought
```rust
self.execute_private(transaction, TransactOptions::with_no_tracing(), block)
```

Reply:
I need to return result.result in this method and abort execution on the first line, if result is not calculated